### PR TITLE
Step 7.2: Pre-Shell Critical Items (F11, F12, F14)

### PR DESCRIPTION
**Pull request by tnaum-ms**
## Step 7.2: Pre-Shell Critical Items

Completes pre-shell cleanup items deferred from Steps 6–7. These harden the playground foundation before the interactive shell (Step 9).

#### 7.2.1: `@microsoft/documentdb-vscode-shell-runtime` Package Extraction (P0)

Extracts the `@mongosh` wrapper into a standalone workspace package with a clean `DocumentDBShellRuntime` API. Both the query playground and the future interactive shell consume this.

- **`DocumentDBShellRuntime`** — public API: `evaluate(code, dbName) → ShellEvaluationResult`
- **`DocumentDBServiceProvider`** — extends `NodeDriverServiceProvider` with DocumentDB metadata
- **`CommandInterceptor`** — pre-eval routing for `help`/`help()`
- **`ResultTransformer`** — normalizes @mongosh ShellResult (cursor unwrap, array subclass normalization)
- **`HelpProvider`** — DocumentDB-specific help text (verified commands only)
- Integrated into `PlaygroundEvaluator.ts` and `playgroundWorker.ts` (replaced ~130 lines of inline code)
- 32 unit tests

#### 7.2.2: `@mongosh` Main Bundle Extraction — F14 (P1)

Removes all `@mongosh` modules from the main extension bundle by routing help commands through the worker.

- **main.js: 12.76 MB → 6.79 MB (−47%)**
- `@mongosh` is now only in the worker bundle (`playgroundWorker.js`)
- Zero `@mongosh` imports in the main bundle

#### 7.2.3: Wire `displayBatchSize` from Settings — F11 (P1)

Wires the `documentDB.shell.batchSize` setting end-to-end so users can control how many documents are returned per cursor iteration.

- Setting renamed: `documentDB.mongoShell.batchSize` → `documentDB.shell.batchSize` (type: `integer`)
- Read per-eval (not at init time) so changes take effect on next run
- Moved `displayBatchSize` from `init` to `eval` message in worker protocol
- Uses `instanceState.displayBatchSizeFromDBQuery` directly (not `config.set()` which silently fails without an evaluation listener)
- Result truncation hint uses `cursorHasMore` from @mongosh (with `displayBatchSize` length heuristic as fallback)
- `PlaygroundDiagnostics` provider: warns when `.limit(N)` exceeds batch size, with comment/string-aware detection and "Open Settings" quick-fix

#### 7.2.4: Help Text Accuracy — F12 (P2)

Audits help text to only list verified-working DocumentDB commands.

- Removed `db.adminCommand({...})` (most admin commands unsupported)
- Marked `db.runCommand({...})` as "limited support"
- 2 new tests for the above

#### 7.2.5: Structure-Aware Block Detection — F13 (P2, Skipped)

Evaluated 4 options (blank-line, bracket-aware, semicolon-terminated, hybrid). Decided to keep the current blank-line splitting — simple, predictable, and the interactive shell will use a different input model.

#### Additional Fixes

**Hover provider:** Fixed `console.log` showing `$log` operator docs — when the character before the hovered word is `.` (property access), the `$`-prefix operator lookup is now suppressed.

**cursorHasMore threading:** Added `cursorHasMore?: boolean` to all result types, extracted in `ResultTransformer` before normalizing cursor results, threaded through worker IPC. Formatter shows truncation hint only when the cursor actually has more documents. Worker has a fallback check on raw printable for robustness.

### Test Results

- Shell-runtime package: 41 tests passing
- Playground completions: 84 tests passing (17 hover/edge-case TDD tests)
- All existing tests unaffected

### CODE REVIEW

File: src/extensionVariables.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export namespace ext {
         export const shellPath = 'documentDB.mongoShell.path';
         export const shellArgs = 'documentDB.mongoShell.args';
         export const shellTimeout = 'documentDB.mongoShell.timeout';
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/extensionVariables.ts:68]: Investigated: the old key (`documentDB.mongoShell.batchSize`) was declared in `package.json` but **never wired at runtime** — `PlaygroundEvaluator.ts` on `feature/shell-integration` has `// TODO(F11)` and hardcodes `displayBatchSize: 50`. No user could have had a working customization of this setting. The rename is safe. No action taken.

File: src/documentdb/playground/PlaygroundDiagnostics.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                vscode.CodeActionKind.QuickFix,
+            );
+            toArrayAction.diagnostics = [diagnostic];
+            toArrayAction.isPreferred = true;
+            toArrayAction.edit = new vscode.WorkspaceEdit();
+
+            // Find the end of the statement — insert .toArray() right after .limit(N)
+            const insertPosition = diagnostic.range.end;
+            toArrayAction.edit.insert(document.uri, insertPosition, '.toArray()');
+            actions.push(toArrayAction);
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/PlaygroundDiagnostics.ts]: Fixed in ea4d35cd. Removed the `.toArray()` quick-fix entirely (it inserted at the wrong position in chained calls). Only the "Open Settings" quick-fix remains.

File: src/documentdb/playground/PlaygroundDiagnostics.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+const DIAGNOSTIC_SOURCE = 'DocumentDB';
+
+/**
+ * Pattern to detect `.limit(N)` calls in playground code.
+ * Captures the numeric argument and the full match range.
+ */
+const LIMIT_CALL_PATTERN = /\.limit\(\s*(\d+)\s*\)/g;
+
+/**
+ * Provides diagnostics and code actions for DocumentDB playground files.
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/PlaygroundDiagnostics.ts:21]: Fixed in ea4d35cd. Added comment/string range detection to suppress false positives. Also skips matches on lines that already contain `.toArray()`. Used a lightweight scanner rather than acorn — sufficient for this use case without adding a parsing dependency to the diagnostic path.

File: src/documentdb/playground/resultFormatter.ts [hunk 4]
```diff
@@ -44,6 +48,20 @@ export function formatResult(result: ExecutionResult, code: string, connection:
...
+    if (
+        result.type === 'Cursor' &&
+        Array.isArray(unwrapped) &&
+        displayBatchSize !== undefined &&
+        unwrapped.length === displayBatchSize
+    ) {
+        lines.push(
+            `// ${l10n.t('Showing first {0} documents (batch size). To change: Settings → documentDB.shell.batchSize', displayBatchSize)}`,
+        );
+    }
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/resultFormatter.ts]: Fixed in 9dcac215. The truncation hint now uses `result.cursorHasMore === true` instead of the `length === displayBatchSize` heuristic. No more false hints when a query happens to return exactly `batchSize` documents.

File: packages/documentdb-shell-runtime/src/ResultTransformer.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        // Cursor result wrapped as { cursorHasMore, documents }
+        if (
+            type === 'Cursor' &&
+            typeof printable === 'object' &&
+            printable !== null &&
+            'documents' in printable &&
+            Array.isArray((printable as { documents?: unknown }).documents)
+        ) {
+            return (printable as { documents: unknown[] }).documents;
+        }
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply packages/documentdb-shell-runtime/src/ResultTransformer.ts:57]: Fixed in 9dcac215. Added `cursorHasMore?: boolean` to `ShellEvaluationResult`. `ResultTransformer` now extracts it from the `{ cursorHasMore, documents }` wrapper before normalizing to a plain array. Threaded through worker IPC (`SerializableExecutionResult`) and `ExecutionResult`. 4 new tests cover the extraction.

File: package.json [hunk 6]
```diff
@@ -1129,10 +1130,11 @@
             "description": "The duration allowed (in seconds) for the Mongo shell to execute a command. Default value is 30 seconds.",
             "default": 30
           },
-          "documentDB.mongoShell.batchSize": {
+          "documentDB.shell.batchSize": {
             "type": "number",
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply package.json]: Fixed in 1f24492d. Changed `"type": "number"` to `"type": "integer"`.

File: src/documentdb/playground/resultFormatter.ts [hunk 7]
```diff
@@ -44,6 +48,22 @@ export function formatResult(result: ExecutionResult, code: string, connection:
...
+        const truncated =
+            result.cursorHasMore === true ||
+            (result.cursorHasMore === undefined &&
+                displayBatchSize !== undefined &&
+                unwrapped.length === displayBatchSize);
+        if (truncated) {
+            lines.push(
+                `// ${l10n.t('Showing first {0} documents (batch size). To change: Settings → documentDB.shell.batchSize', unwrapped.length)}`,
+            );
+        }
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/resultFormatter.ts]: acceptable, that's a fall back solution

[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/resultFormatter.ts]: Fixed in eb426231. Dropped the `displayBatchSize` fallback heuristic entirely. The truncation hint now only shows when `result.cursorHasMore === true`. Confirmed via debug tracing that the `transformer` path reliably provides `cursorHasMore` from @mongosh's `CursorIterationResult`.

File: src/documentdb/playground/playgroundWorker.ts [hunk 8]
```diff
@@ -142,134 +158,52 @@ async function handleInit(msg: Extract<MainToWorkerMessage, { type: 'init' }>):
...
-        printableValue = Array.from(shellResult.printable as unknown[]);
+    // Proactively extract cursorHasMore before serialization.
+    // The ResultTransformer may have already extracted it from the { cursorHasMore, documents }
+    // wrapper, but in some @mongosh/vm contexts the property survives on the normalized
+    // printable array as an own property. Check both the runtime result and the raw printable.
+    let cursorHasMore = result.cursorHasMore;
+    if (cursorHasMore === undefined && result.type === 'Cursor') {
+        const p = result.printable;
+        if (typeof p === 'object' && p !== null && 'cursorHasMore' in p) {
+            cursorHasMore = Boolean((p as { cursorHasMore: unknown }).cursorHasMore);
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/playgroundWorker.ts]: Fixed in eb426231. Now uses `typeof fallbackValue === 'boolean'` guard instead of `Boolean()` coercion. Non-boolean values are left as `undefined`.

File: src/documentdb/playground/PlaygroundDiagnostics.ts [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+import * as l10n from '@vscode/l10n';
+import * as vscode from 'vscode';
+import { ext } from '../../extensionVariables';
+import { getBatchSizeSetting } from '../../utils/workspacUtils';
+
+const PLAYGROUND_LANGUAGE_ID = 'documentdb-playground';
+const DIAGNOSTIC_SOURCE = 'DocumentDB';
+
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/playground/PlaygroundDiagnostics.ts]: Fixed in eb426231. Now imports `PLAYGROUND_LANGUAGE_ID` from `./constants` instead of hardcoding the string.
